spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/task/TaskLauncher.java

```diff
@@ -103,4 +104,17 @@
 	 * @return the deployer environment info
 	 */
 	RuntimeEnvironmentInfo environmentInfo();
+
+	/**
+	 * Implementations may limit the number of concurrent task executions.
+	 *
+	 * @return the maximum concurrent task executions allowed.
+	 */
+	int getMaximumConcurrentTasks();
+
+	/**
+	 *
+	 * @return the count of currently running task executions if known.
+	 */
+	int getRunningTaskExecutionCount();
```

Comment:
Funny, I thought maybe the method name `getCurrentlyRunningTaskExecutionCount` or `getCurrentTaskExecution` would be a slightly better name as it aligns with the javadoc description.  I think all the methods have an implied 'if known' to them.

Reply:
add default method to throw `UnsupportedOperationException`